pkg/skuba/actions/node/upgrade/apply.go

```diff
@@ -107,6 +115,17 @@ func Apply(target *deployments.Target) error {
 			if err := target.Apply(nil, "kubernetes.restart-services"); err != nil {
 				return err
 			}
+			if err := target.Apply(nil, "kubernetes.wait-for-kubelet"); err != nil {
+				return err
```

Comment:
What happens with the lock if the Kubelet failed to start? I wonder if we should then free the lock and annotate the node somehow :thinking:

Reply:
the lock will still be present, as the `apply` command will exit here. i'm not sure how to best continue in such a case, but if kubelet is not able to register the node, there's no chance you can perform anything at that stage. for best effort we can enhance the logging and print out a message to be aware of that i'd say

Reply:
i've added some logging now